dcc-submission-core/src/main/java/org/icgc/dcc/submission/dictionary/model/FileSchema.java

```diff
@@ -336,4 +333,40 @@ public boolean apply(Field field) {
     });
   }
 
+  /**
+   * Compares according to {@link Relation}. If {@code this} is a parent of the {@code ot}
+   * @return -1 if {@code this} is a parent of the {@code other}<br>
+   * 0 if they are equal<br>
+   * 1 if {@code this} is a child of the {@code other}
+   */
+  @Override
+  public int compareTo(FileSchema other) {
```

Comment:
I'm not sure `Comparable` is the right mechanism here. This is a very context specific ordering that probably would be better served in a `DictionaryTopologicalComparator` implementation.